pkg/types/entitlement/entilement.go

```diff
@@ -0,0 +1,37 @@
+package entitlement
+
+import (
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"google.golang.org/protobuf/proto"
+)
+
+type EntitlementOption func(*v2.Entitlement)
```

Comment:
What do you think of moving all of the `New*` functions into these smaller type specific packages so we wouldn't have any of this in the `sdk/` package?

Reply:
Yeah, I think that makes more sense. I'll get a commit added.